fx.go

```diff
@@ -19,3 +19,149 @@
 // THE SOFTWARE.
 
 package fx // import "go.uber.org/fx"
+
+import (
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+
+	"go.uber.org/dig"
+	"go.uber.org/fx/config"
+	"go.uber.org/fx/metrics"
+	"go.uber.org/fx/ulog"
+	"go.uber.org/zap"
+)
+
+// Component is a function that initializes and returns objects to be shared across the code
+type Component interface{}
+
+// Module is a building block or primary function performed by a service built on fx
+type Module interface {
+	Name() string
+	Constructor() []Component
```

Comment:
To Glib's question:

Right now, outbounds are very specific to the client i.e. you need the outbound specific import and the type is also specific to the outbound client so the use case doesn't exist yet.
It is possible that in future there is a generic way to create outbounds.

If so, there is nothing that limits this to config. It could be any other dependency as well or the module could have their own config library based on which they want to do this. If we really want to support this, we should talk about sending the dig graph in to the constructor or to the module construction for more flexibility.